notebook/notebookapp.py

```diff
@@ -1286,11 +1288,11 @@ def _confirm_exit(self):
         # set it back to original SIGINT handler
         # use IOLoop.add_callback because signal.signal must be called
         # from main thread
-        ioloop.IOLoop.current().add_callback(self._restore_sigint_handler)
+        self.io_loop.add_callback_from_signal(self._restore_sigint_handler)
     
     def _signal_stop(self, sig, frame):
         self.log.critical(_("received signal %s, stopping"), sig)
-        ioloop.IOLoop.current().stop()
+        self.io_loop.add_callback(self.io_loop.stop)
```

Comment:
This one should be using the `_from_signal` version of the method, though, as this function is registered directly as a signal handler.